snapraid-aio-script.sh

```diff
@@ -455,21 +469,24 @@ function prepare_mail() {
 function send_mail(){
   # Format for markdown
   sed_me "s:$:  :" "$TMP_OUTPUT"
-  sed  "/^Running TOUCH job to timestamp/,/^\TOUCH finished/{/^Running TOUCH job to timestamp/!{/^TOUCH finished/!d}}; /^###SnapRAID DIFF/,/^\DIFF finished/{/^###SnapRAID DIFF/!{/^DIFF finished/!d}}" $TMP_OUTPUT | $MAIL_BIN -a 'Content-Type: text/html' -s "$SUBJECT" "$EMAIL_ADDRESS" < <(python -m markdown)  
+  sed  "/^Running TOUCH job to timestamp/,/^\TOUCH finished/{/^Running TOUCH job to timestamp/!{/^TOUCH finished/!d}}; /^###SnapRAID DIFF/,/^\DIFF finished/{/^###SnapRAID DIFF/!{/^DIFF finished/!d}}" "$TMP_OUTPUT" | $MAIL_BIN -a 'Content-Type: text/html' -s "$SUBJECT" "$EMAIL_ADDRESS" < <(python -m markdown)
 }
 
 function send_mail_verbose(){
   # Format for markdown
   sed_me "s:$:  :" "$TMP_OUTPUT"
-  $MAIL_BIN -a 'Content-Type: text/html' -s "$SUBJECT" "$EMAIL_ADDRESS" < $TMP_OUTPUT < <(python -m markdown)  
+  $MAIL_BIN -a 'Content-Type: text/html' -s "$SUBJECT" "$EMAIL_ADDRESS" < "$TMP_OUTPUT" < <(python -m markdown)
 }
 
-#Due to how process substitution and newer bash versions work, this function stops the output stream which allows wait stops wait from hanging on the tee process.
-#If we do not do this and use normal 'wait' the processes will wait forever as newer bash version will wait for the process substitution to finish.
-#Probably not the best way of 'fixing' this issue. Someone with more knowledge can provide better insight.
+# Due to how process substitution and newer bash versions work, this function
+# stops the output stream which allows wait stops wait from hanging on the tee
+# process. If we do not do this and use normal 'wait' the processes will wait
+# forever as newer bash version will wait for the process substitution to
+# finish. Probably not the best way of 'fixing' this issue. Someone with more
+# knowledge can provide better insight.
 function close_output_and_wait(){
-  exec >&$out 2>&$err
-  wait $(pgrep -P "$$")
+  exec >&"$out" 2>&"$err"
+  wait "$(pgrep -P $$)"
```

Comment:
Similar story like before: the error disappears if the change is reverted to

`wait $(pgrep -P "$$")`

But I don't know if it's the way you like it.

Reply:
Allright, I _think_ the correct one is:
`wait $"(pgrep -P "$$")"`

With this I don't get the error anymore and everything is smartly doublequoted.

Reply:
Oh, yeah. Sorry I noticed that before but fixed in another diff outside of this PR. I believe the issue is that `pgrep` returns nothing and then `wait` gets called with no args. 
```
bash -c 'wait ""'
bash: line 0: wait: `': not a pid or valid job spec

bash -c 'wait'
```
The warning is safe to ignore for now, but I'll add the fix for this.

Reply:
Allright, I will merge this to master. I can see you're working on more stuff, so I'll wait before making changes (or a release).